### DISCUSSION TITLE
How can I deploy Avalonia on an Apache webserver?

### DISCUSSION THREAD

**Discussion by imadiz** [AvaloniaUI/Avalonia#14214] [Q&A]
Hello!

I recently stumbled upon Avalonia and I started writing little projects in it, but when I wanted to publish the browser project to a folder from Visual Studio, it simply didn't create any files.

Then I started digging through forums and discussions, and found a lot of differents parts of different solutions, but they didn't work/I didn't understand them.

The browser project works if I launch it from Visual Studio, but when I publish from cmd with dotnet it creates an AppBundle folder and when I try to use it with Apache, it just shows the default sample app. (And a bunch of errors)

![image](https://github.com/AvaloniaUI/Avalonia/assets/70607543/cc709a1f-4427-49bf-846f-7d9f460f5437)

Edit: The files are there, VS Code can find it by Ctrl Clicking.

Just to clarify, I have little to no experience with web developing, I only work with C#(and ClickOnce).

Also, english is a second language for me, so sorry if it's a little weird at times.

Can I please get some help for a solution.

Edit: My only lead is that running the project from cmd seems to use the .Browser.runtimeconfig.json file.

![image](https://github.com/AvaloniaUI/Avalonia/assets/70607543/64098f76-989e-488e-950a-3b0fa5c9e223)

Thanks for the help!

**Comment by Yarchikkk** [comment]
Hi, also trying to figure out how to upload the project as a website to hosting, were you able to figure it out?

**Comment by imadiz** [comment]
Hey man,
If youre still interested, I found a dotnet CMD tool called dotnet serve.
It runs Avalonia pages fine, but you need to attach a domain name to its
hosting IP, and maybe make it a service on a server machine. I've got only
this far, but didnt actually do the thing, routing and IPs are not my thing.
If you accomplish something and have the time, I would appreciate if you'd
write about it!
Thanks!
Good luck to you!
